kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/Resource.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.pulsar.handlers.kop.security.auth;
+
+
+import lombok.Getter;
+
+/**
+ * The Authorization resource.
+ */
+public class Resource {
```

Comment:
We can add `@Getter` as class annotation to avoid adding `@Getter` for each field. In addition, use `@AllArgsConstructor` here. If you prefer to use factory method `Resource#of` than the constructor, please make the constructor private.